pkg/bpf/map_linux_test.go

```diff
@@ -384,6 +387,97 @@ func (s *BPFPrivilegedTestSuite) TestDump(c *C) {
 	})
 }
 
+func (s *BPFPrivilegedTestSuite) TestDumpReliablyWithCallback(c *C) {
+	maxEntries := uint32(256)
+	m := NewMap("cilium_dump_test",
+		MapTypeHash,
+		&TestKey{},
+		int(unsafe.Sizeof(TestKey{})),
+		&TestValue{},
+		int(unsafe.Sizeof(TestValue{})),
+		int(maxEntries),
+		BPF_F_NO_PREALLOC,
+		0,
+		ConvertKeyValue,
+	).WithCache()
+	_, err := m.OpenOrCreate()
+	if err != nil {
```

Comment:
This if branch is not needed

Reply:
Would you mind elaborating on that?

Reply:
Just writing:

```
_, err := m.OpenOrCreate()
c.Assert(err, IsNil)
```

is enough. `c.Assert(err, IsNil)` is actually checking whether `err` is not `nil`. It's in fact a shorter version of doing `if err != nil { t.Errorf("unexpected error: %v", err) }` with standard library tests.

Reply:
Sorry, I must be tired :)